packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -214,9 +214,82 @@ export enum ContainerMessageType {
 	IdAllocation = "idAllocation",
 }
 
+/**
+ * How should an older client handle an unrecognized remote op type?
+ *
+ * @internal
+ */
+export type CompatModeBehavior =
+	/** Ignore the op. It won't be persisted if this client summarizes */
+	| "Ignore"
+	/** Fail processing immediately. (The container will close) */
+	| "FailToProcess";
+
+/**
+ * All the info an older client would need to know how to handle an unrecognized remote op type
+ *
+ * @internal
+ */
+export interface IContainerRuntimeMessageCompatDetails {
+	/** How should an older client handle an unrecognized remote op type? */
+	behavior: CompatModeBehavior;
+}
+
+/** If a message doesn't specify compat behavior, it is not suitable for old clients to process */
+const defaultCompatDetails: IContainerRuntimeMessageCompatDetails = { behavior: "FailToProcess" };
+
 export interface ContainerRuntimeMessage {
-	contents: any;
 	type: ContainerMessageType;
+	contents: any;
+	/** @see defaultCompatDetails */
+	compatDetails?: IContainerRuntimeMessageCompatDetails;
```

Comment:
Maybe too early to ask, but will this be part of the protocol (ie the server will relay it) or will we pack it further in the content?

Reply:
Nope, it's buried inside op content